controllers/core/openstackcontrolplane_controller.go

```diff
@@ -159,10 +159,14 @@ func (r *OpenStackControlPlaneReconciler) Reconcile(ctx context.Context, req ctr
 			condition.UnknownCondition(corev1beta1.OpenStackControlPlaneKeystoneAPIReadyCondition, condition.InitReason, corev1beta1.OpenStackControlPlaneKeystoneAPIReadyInitMessage),
 			condition.UnknownCondition(corev1beta1.OpenStackControlPlanePlacementAPIReadyCondition, condition.InitReason, corev1beta1.OpenStackControlPlanePlacementAPIReadyInitMessage),
 			condition.UnknownCondition(corev1beta1.OpenStackControlPlaneGlanceReadyCondition, condition.InitReason, corev1beta1.OpenStackControlPlaneGlanceReadyInitMessage),
-			condition.UnknownCondition(corev1beta1.OpenStackControlPlaneCinderReadyCondition, condition.InitReason, corev1beta1.OpenStackControlPlaneCinderReadyInitMessage),
 			condition.UnknownCondition(corev1beta1.OpenStackControlPlaneNovaReadyCondition, condition.InitReason, corev1beta1.OpenStackControlPlaneNovaReadyInitMessage),
 		)
 
+		// Initialize optional service conditions if enabled
+		if instance.Spec.Cinder.Enabled {
+			cl = append(cl, *condition.UnknownCondition(corev1beta1.OpenStackControlPlaneCinderReadyCondition, condition.InitReason, corev1beta1.OpenStackControlPlaneCinderReadyInitMessage))
+		}
```

Comment:
I'm not quite familiar with the standard but does it make sense to present the condition with "disabled" instead of not exposing it at all ?

Reply:
I think my preference would be to just leave it out of the status entirely. Few reasons for that, but mainly because the purpose of the status is to report the outcome of our reconcile attempts. Since we shouldn't really try to reconcile a disabled service at all, there's not really a reason to include that service in the status report. That's at least my opinion on the decision to not include it.

Reply:
Fair enough. Let me keep this open in case others want to chime in

Reply:
I don't think I have a preference for this, at least at the moment. Just wondering if we should have the same across all the services.

Reply:
I think I still preference leaving it out to avoid confusion. I'll push another update to this PR with a implementation proposal.

Reply:
So, ultimately the outcome here is that we will end up marking the service as Ready, with a message in the status field that the Service is disabled.

I understand @gibizer point around having a message to explain it's disabled, when we can also see it's disabled in the spec. I think probably from a user perspective, they might be wondering why the service has been marked as Ready, and having the message there probably helps to clarify. So that is implemented in the cinder.go `ReconcileCinder()` function, during the service enabled check.